platform/backend/repository/testbed/src/test/kotlin/io/hamal/repository/SnippetTest.kt

```diff
@@ -0,0 +1,349 @@
+            )
+
+            with(result) {
+                assertThat(id, equalTo(SnippetId(1)))
+                assertThat(groupId, equalTo(GroupId(1)))
+                assertThat(name, equalTo(SnippetName("TestSnippet2")))
+                assertThat(value, equalTo(CodeValue("1 + 1")))
+                assertThat(inputs, equalTo(SnippetInputs(MapType(mutableMapOf("answer" to NumberType(42))))))
+                assertThat(accountId, equalTo(AccountId("123")))
+            }
+
+            verifyCount(1)
+        }
+
+        @TestFactory
+        fun `Updates code multiple times`() = runWith(SnippetRepository::class) {
+            createSnippet(
+                snippetId = SnippetId(1),
+                groupId = GroupId(1),
+                name = SnippetName("TestSnippet")
+            )
+
+            repeat(13) { iteration ->
+                val result = update(
+                    SnippetId(1), UpdateCmd(
+                        id = CmdId(iteration + 2),
+                        name = SnippetName("TestSnippet$iteration"),
+                        value = CodeValue("40 + $iteration"),
+                        inputs = SnippetInputs(MapType(mutableMapOf("answer" to NumberType(42))))
+                    )
+                )
+
+                with(result) {
+                    assertThat(id, equalTo(SnippetId(1)))
+                    assertThat(groupId, equalTo(GroupId(1)))
+                    assertThat(value, equalTo(CodeValue("40 + $iteration")))
+                    assertThat(accountId, equalTo(AccountId("123")))
+                    assertThat(name, equalTo(SnippetName("TestSnippet$iteration")))
+                }
+            }
+            verifyCount(1)
+        }
+
+    }
+
+    @Nested
+    inner class FindTest {
+
+
+        @TestFactory
+        fun `Find func by id`() = runWith(SnippetRepository::class) {
+            createSnippet(
+                snippetId = SnippetId(1),
+                groupId = GroupId(1),
+                name = SnippetName("TestSnippet"),
+                value = CodeValue("1 + 1")
+            )
+
+            with(find(SnippetId(1))!!) {
+                assertThat(id, equalTo(SnippetId(1)))
+                assertThat(groupId, equalTo(GroupId(1)))
+                assertThat(name, equalTo(SnippetName("TestSnippet")))
+                assertThat(value, equalTo(CodeValue("1 + 1")))
+                assertThat(accountId, equalTo(AccountId("123")))
+
+            }
+        }
+
+        @TestFactory
+        fun `Tries to get snippet by id but does not exist`() = runWith(SnippetRepository::class) {
+            createSnippet(
+                snippetId = SnippetId(1),
+                groupId = GroupId(1),
+                name = SnippetName("TestSnippet")
+            )
+
+            val exception = assertThrows<NoSuchElementException> {
+                get(SnippetId(111111))
+            }
+            assertThat(exception.message, equalTo("Snippet not found"))
+        }
+    }
```

Comment:
This must be part of GetTest - as you are testing the get method not the find method! We need a negative test for find as well in the end we should have 4 tests for get and find

FindTest{
 // we find the snippet
// we dont find the snippet -- null - no exception
}

GetTest{
// we get the snippet
// we are getting an exception
}